blocks/identity-block/components/social-sign-on/utils/useSocialSignIn.js

```diff
@@ -4,30 +4,59 @@ import { GoogleSignInContext } from "./googleContext";
 import useOIDCLogin from "../../../utils/useOIDCLogin";
 import validateURL from "../../../utils/validate-redirect-url";
 
-import {SIGN_UP} from "../constants";
+import { SIGN_UP } from "../constants";
 
-function useSocialSignIn(redirectURL, isOIDC, socialSignOnIn, onError = () => {}) {
+function useSocialSignIn(
+	redirectURL,
+	isOIDC,
+	socialSignOnIn,
+	onError = () => {},
+	customButtons = true,
+	fromSocialSignOnBlock = true,
+) {
 	const { Identity } = useIdentity();
 	const { isGoogleLoaded } = useContext(GoogleSignInContext);
 	const [config, setConfig] = useState(() => Identity?.configOptions ?? {});
+
+	const [isFBInitialized, setIsFBInitialized] = useState(false);
+	const [isFBEventSubscribed, setIsFBEventSubscribed] = useState(false);
+	const [updateIdentities, setUpdateIdentities] = useState(false);
+
 	const { loginByOIDC } = useOIDCLogin();
 
 	useEffect(() => {
 		window.onFacebookSignOn = async () => {
 			try {
-				await Identity.facebookSignOn();
+				Identity.facebookSignOn().then(() => {
+					if (customButtons) {
+						setIsFBInitialized(true);
+					}
+				});
 
 				if (isOIDC) {
 					loginByOIDC();
-				} else {
-					const validatedURL = validateURL(redirectURL);
-					window.location = validatedURL;
+				} else if (fromSocialSignOnBlock) {
+					Identity.isLoggedIn().then((isLoggedIn) => {
+						if (isLoggedIn) {
+							const validatedURL = validateURL(redirectURL);
+							window.location = validatedURL;
```

Comment:
There's no need to assign it to a constant then immediately set the location to that constant...
```suggestion
							window.location = validateURL(redirectURL);
```